actions/vmwarelib/guest.py

```diff
@@ -0,0 +1,169 @@
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyVmomi import vim  # pylint: disable-msg=E0611
+import eventlet
+import os
+import requests
+import sys
+
+
+def auth_username_password(username, password):
+    return vim.vm.guest.NamePasswordAuthentication(username=username,
+                                                   password=password)
+
+
+def dir_create(content, vm, creds, prefix, suffix):
+    """
+    Create a "temporary" directory (using a unique name).
+
+    Returns:
+    - the full pathname to the "temporary" directory in the guest
+    """
+    file_manager = content.guestOperationsManager.fileManager
+    return file_manager.CreateTemporaryDirectoryInGuest(vm, creds,
+                                                        prefix, suffix)
+
+
+def dir_delete(content, vm, creds, directory, recursive=False):
+    """
+    Delete a directory with a recursive option.
+    """
+    file_manager = content.guestOperationsManager.fileManager
+    file_manager.DeleteDirectoryInGuest(vm, creds,
+                                        directory, recursive)
+
+
+def file_create(content, vm, creds, directory, prefix, suffix):
+    """
+    Create a "temporary" file in a directory (using a unique name).
+
+    Returns:
+    - the full pathname to the "temporary" file
+    """
+    file_manager = content.guestOperationsManager.fileManager
+    return file_manager.CreateTemporaryFileInGuest(vm, creds,
+                                                   prefix, suffix,
+                                                   directory)
+
+
+def file_delete(content, vm, creds, guestfile):
+    """
+    Delete a file inside the guest.
+    """
+    file_manager = content.guestOperationsManager.fileManager
+    file_manager.DeleteFileInGuest(vm, creds, guestfile)
+
+
+def file_read(content, vm, creds, guest_file):
+    """
+    Read the contents of a file on the guest.
+
+    Returns:
+    - the contents of the file
+    """
+    file_manager = content.guestOperationsManager.fileManager
+    dl_url = file_manager.InitiateFileTransferFromGuest(vm, creds, guest_file)
+    return (requests.get(dl_url.url, verify=False)).text
+
+
+def file_upload(content, vm, creds, guest_directory, local_file):
+    """
+    Upload a file into a directory in the guest.
+
+    Args:
+    - local_file is a path relative to the 'packs' directory
+
+    Returns:
+    - full path to uploaded file in the guest
+    """
+    packdir =\
```

Comment:
I'm not sure it's a good idea relying on the file path being relative to the packs directory, or if it's even the best approach for users to put that content in their own pack directories. Feels like it would be better for users to provide an explicit path.

Reply:
It's definitely better to have people put their scripts into their own packs with their own actions or workflows that leverage ``vsphere.guest-script-run``.  We don't want people adding their scripts to the vsphere pack.  Alternatively, sure, someone could just provide a docker volume with a library of scripts to use I suppose.  I suppose a full path is a better choice.